src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
```

Comment:
I added comments to the .h file to clarify: callers of these functions regularly check for evenness, so I decided to make that requirement explicit and remove the even-checks from the call sites.   I will add a TODO to look at the API issues later.